packages/loot-core/src/shared/util.test.ts

```diff
@@ -37,31 +37,37 @@ describe('utility functions', () => {
   test('number formatting works with comma-dot format', () => {
     setNumberFormat({ format: 'comma-dot', hideFraction: false });
     let formatter = getNumberFormat().formatter;
+    // @ts-expect-error TS type for format accepts only a number
     expect(formatter.format('1234.56')).toBe('1,234.56');
```

Comment:
hm… maybe port the tests to use number literals instead and throw an error in the API if it’s passed a string to check that that doesn’t happen? (or loosen the API to take strings)

Reply:
Well, [the spec says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/format) `format()` accepts "a Number, BigInt, or string, to format". Here we are using TS `Intl.NumberFormat` type definition that however does only accept number type (they seem to [not want to change that](https://github.com/microsoft/TypeScript/issues/52124) ). I feel `@ts-expect-error` is ok under these circumstances, as the implementation is correct and it is just the builtin type that is too strict. But happy to change the test: your call.

Reply:
I think the builtin type is reasonable and it makes sense to explicitly handle the coercion to `number` here to ensure that we don’t accidentally pass in a user-entered string which should be parsed using the current number style.